src/transformations/utils/build-super-calls-for-chain.ts

```diff
@@ -83,20 +84,59 @@ export function buildSuperCallsForChain(
   const linearizedCtorCalls: string[] = [];
   // this is where we store the parents of uninitialized parents if any
   const notInitializable = new Set<number>();
+  const argsValues: Record<number, Expression> = {};
+  const parentArgsValues: Record<number, Expression[]> = {};
```

Comment:
For many (most) numbers the value will be undefined so we want to express this in the type:

```suggestion
  const literalArgsValues: Record<number, Expression | undefined> = {};
  const argsValues: Record<number, Expression[] | undefined> = {};
```

Can you also add comments explaining what the `number` values are for each record?